atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceIntegrationTest.java

```diff
@@ -300,6 +308,47 @@ public void deletionTakesPlaceAtFreshTimestamp() throws Exception {
         assertThat(results).doesNotContainKey(CELL);
     }
 
+    @Test
+    public void testRangeLoader() throws Exception {
+        TableReference tableReference =
+                TableReference.createFromFullyQualifiedName("test." + RandomStringUtils.randomAlphanumeric(16));
+        keyValueService.createTable(tableReference, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        byte[] data = PtBytes.toBytes("data");
+        byte[] moreData = PtBytes.toBytes("data2");
+
+        Cell row_1_cell = Cell.create(PtBytes.toBytes("row_1"), PtBytes.toBytes("column"));
+
+        keyValueService.putWithTimestamps(tableReference, ImmutableListMultimap.of(row_1_cell, Value.create(data, 8L)));
+        keyValueService.putWithTimestamps(tableReference, ImmutableListMultimap.of(row_1_cell, Value.create(moreData, 88L)));
+
+        TableMetadata tableMetadata = KeyValueServices.getTableMetadataSafe(keyValueService, tableReference);
+
+        Set<byte[]> allColumnNames = tableMetadata.getColumns().getAllColumnNames();
+        Assert.assertEquals(allColumnNames.size(), 1);
+
+        // case 1: single column table, columns specified in the rangeRequest
+        RangeRequest rangeRequest = RangeRequest.builder().retainColumns(allColumnNames).build();
+        ClosableIterator<RowResult<Value>> results_1 = keyValueService.getRange(tableReference,
+                rangeRequest,
+                (-89L));
+
+
+        // single column table, columns are not specified in the rangeRequest
+        rangeRequest = RangeRequest.builder().build();
+        ClosableIterator<RowResult<Value>> results_2 = keyValueService.getRange(tableReference,
+                rangeRequest,
+                (-89L));
+
+        assertThat(results_1.hasNext());
+        assertThat(results_2.hasNext());
+
+        byte[] contents_1 = results_1.next().getOnlyColumnValue().getContents();
+        byte[] contents_2 = results_2.next().getOnlyColumnValue().getContents();
+
+        Assert.assertEquals(contents_1, contents_2);
```

Comment:
unless I'm mistaken, isn't it hard to test this performance optimisation or is the claim that if you didn't add what you added, then this test would fail because it would return too much data?

Reply:
if so, can we have a comment?